actor/src/macros.rs

```diff
@@ -0,0 +1,122 @@
+                where A: $trait_name
+            {
+                let mut shutdown = Box::pin(shutdown);
+                loop {
+                    let delivery = tokio::select!{
+                        // Check if we should shutdown, this is a graceful shutdown as we do not cancel
+                        // any existing handle futures.
+                        //
+                        // This is an intentional design decision to avoid having to deal with cancel
+                        // safety in handle implementations.
+                        _ = &mut shutdown => {break;}
+                        Some(delivery) = receiver.recv() => {delivery}
+                    };
+                    match delivery {
+                        $crate::DeliverOp::Send(traced) => {
+                            match traced.value {
+                                $(
+                                    $enum_name::$variant_name(m, respond_to) => {
+                                        if respond_to.send($crate::tracing::Instrument::instrument(
+                                            actor.handle(m),
+                                            traced.span,
+                                        ).await).is_err() {
+                                            ::tracing::warn!("failed to send message to actor");
+                                        }
+                                    }
+                                )*
+                            }
+                        }
+                        $crate::DeliverOp::Notify(traced) => {
+                            match traced.value {
+                                $(
+                                    $enum_name::$variant_name(m, _respond_to) => {
+                                        // When notifying we do not care about the response, drop
+                                        // it.
+                                        let _ = $crate::tracing::Instrument::instrument(
+                                            actor.handle(m),
+                                            traced.span,
+                                        ).await;
+                                    }
+                                )*
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        $(
+        impl std::convert::From<($message_type, tokio::sync::oneshot::Sender<<$message_type as $crate::Message>::Result>)> for $enum_name {
+            fn from(value: ($message_type, tokio::sync::oneshot::Sender<<$message_type as $crate::Message>::Result>)) -> Self {
+                Self::$variant_name(value.0, value.1)
+            }
+        }
+        impl std::convert::TryFrom<$enum_name> for $message_type {
+            type Error = ();
+            /// Extracts the message from the envelope.
+            ///
+            /// # Panics
```

Comment:
This isn't true now, is it?

Reply:
good catch, used to be just From hence the comment.